mcs/build/library.make

```diff
@@ -66,6 +78,7 @@ lib_dir = lib
 endif
 
 ifdef LIBRARY_SUBDIR
+profile_library = $(PROFILE)_$(LIBRARY_SUBDIR)_$(LIBRARY)
```

Comment:
We cannot have both `intermediate` and `LIBRARY_SUBDIR` defined at the same time.  This should still be cleaned up, though.